tests/functional/test_install_wheel.py

```diff
@@ -129,6 +129,42 @@ def test_install_wheel_with_target(script, data):
     )
 
 
+@pytest.mark.network
+def test_install_wheel_with_target_and_data_files(script, data):
+    """
+    Test for issue #4092. It will be checked that a data_files specification in
+    setup.py is handled correctly when a wheel is installed with the --target
+    option.
+
+    The setup() for the wheel 'pip_issue_4092-1.0-py2.py3-none-any.whl' is as
+    follows ::
+
+        setup(
+            name='pip_issue_4092',
+            version='1.0',
+            packages=['pip_issue_4092'],
+            data_files=[
+                (r'packages1', ['pip_issue_4092/README.txt']),
+                (r'packages2', ['pip_issue_4092/README.txt'])
+            ]
+        )
+    """
+    script.pip('install', 'wheel')
+    target_dir = script.scratch_path / 'target_4092'
+    package = data.packages.join("pip_issue_4092-1.0-py2.py3-none-any.whl")
```

Comment:
Maybe you could directly create the wheel in this test with something like:
https://github.com/pypa/pip/blob/2713a1d1494056ab4c5490f26f96dd43ab81f500/tests/functional/test_uninstall.py#L166-L176 followed with:

    script.pip('wheel', pkg_path, '--wheel-dir', script.scratch_path)`
    script.pip('install', script.scratch_path / 'pip_issue_4092-1.0-py2.py3-none-any.whl')`

Either that or put the source of your uploaded binary wheel tests/data/packages/pip_issue_4092-1.0-py2.py3-none-any.whl in https://github.com/pypa/pip/tree/master/tests/data/src